agent-connection/src/main/kotlin/com/luxoft/blockchainlab/corda/hyperledger/indy/PythonRefAgentConnection.kt

```diff
@@ -0,0 +1,233 @@
+            true
+        } else {
+            connectionStatus = AgentConnectionStatus.AGENT_CONNECTION_DISCONNECTED
+            false
+        }
+    }
+
+    override fun acceptInvite(invite: String) {
+        if (invite != null && getConnectionStatus() == AgentConnectionStatus.AGENT_CONNECTION_CONNECTED) {
+            sendJson(ReceiveInviteMessage(invite))
+            val invite = webSocket.waitForMessageOfType<InviteReceivedMessage>(MESSAGE_TYPES.INVITE_RECEIVED)
+            sendRequest(invite.key)
+            val response = webSocket.waitForMessageOfType<RequestResponseReceivedMessage>(MESSAGE_TYPES.RESPONSE_RECEIVED)
+            counterParty = IndyParty(response.their_did, invite.endpoint)
+        } else {
+            throw AgentConnectionException("AgentConnection object has wrong state")
+        }
+    }
+
+    fun sendJson(obj: Any) = webSocket.sendJson(obj)
+
+    override fun genInvite(): String {
+        webSocket.sendJson(SendMessage(type = MESSAGE_TYPES.GENERATE_INVITE, id = Instant.now().toEpochMilli().toString()))
+        return webSocket.waitForMessageOfType<ReceiveInviteMessage>(MESSAGE_TYPES.INVITE_GENERATED).invite
+    }
+
+    override fun waitForInvitedParty() {
+        if (getConnectionStatus() == AgentConnectionStatus.AGENT_CONNECTION_CONNECTED) {
+            webSocket.waitForMessageOfType<RequestReceivedMessage>(MESSAGE_TYPES.REQUEST_RECEIVED).also {
+                counterParty = IndyParty(it.did, it.endpoint)
+                sendJson(RequestSendResponseMessage(it.did))
+            }
+        } else {
+            throw AgentConnectionException("Agent is disconnected")
+        }
+    }
+
+    fun sendRequest(key: String) = webSocket.sendJson(SendRequestMessage(key))
+
+    override fun getCounterParty(): IndyParty? {
+        return counterParty
+    }
+
+    override fun sendCredentialOffer(offer: CredentialOffer) = webSocket.sendTypedMessage(offer, counterParty!!)
+
+    override fun receiveCredentialOffer(): CredentialOffer = webSocket.waitForTypedMessage()
+
+    override fun sendCredentialRequest(request: CredentialRequestInfo) = webSocket.sendTypedMessage(request, counterParty!!)
+
+    override fun receiveCredentialRequest(): CredentialRequestInfo = webSocket.waitForTypedMessage()
+
+    override fun sendCredential(credential: CredentialInfo) = webSocket.sendTypedMessage(credential, counterParty!!)
+
+    override fun receiveCredential(): CredentialInfo = webSocket.waitForTypedMessage()
+    override fun sendProofRequest(request: ProofRequest) = webSocket.sendTypedMessage(request, counterParty!!)
+
+    override fun receiveProofRequest(): ProofRequest = webSocket.waitForTypedMessage()
+
+    override fun sendProof(proof: ProofInfo) = webSocket.sendTypedMessage(proof, counterParty!!)
+
+    override fun receiveProof(): ProofInfo = webSocket.waitForTypedMessage()
+}
+
+class AgentWebSocketClient(serverUri: URI) : WebSocketClient(serverUri) {
+    private val log = KotlinLogging.logger {}
```

Comment:
Let's use there same logging system everywhere

Reply:
It is same, what do you mean?

Reply:
We used another one, do we on kotlin logging everywhere in cordentity? Could you double check that no other logging lib is used in other modules of project

Reply:
AFAIK, i`ve replaced everywhere.
https://github.com/hyperledger-labs/cordentity/pull/12/commits/80a2dcb2bec5f2a45e17ba189bd4d1e7df799bfb

Reply:
Let's put it to backlog and do improvement later.
BTW Could you attach a link on each task in comments where its required, please?  I'll solve the conversations than

Reply:
https://github.com/hyperledger-labs/cordentity/projects/1#card-18735700